{"metadata":{"kernelspec":{"language":"python","display_name":"Python 3","name":"python3"},"language_info":{"name":"python","version":"3.10.14","mimetype":"text/x-python","codemirror_mode":{"name":"ipython","version":3},"pygments_lexer":"ipython3","nbconvert_exporter":"python","file_extension":".py"},"kaggle":{"accelerator":"none","dataSources":[{"sourceId":4319733,"sourceType":"datasetVersion","datasetId":2544155}],"dockerImageVersionId":30804,"isInternetEnabled":true,"language":"python","sourceType":"notebook","isGpuEnabled":false}},"nbformat_minor":4,"nbformat":4,"cells":[{"cell_type":"code","source":"import numpy as np\nimport pandas as pd\nfrom sklearn.preprocessing import LabelEncoder\nfrom sklearn.model_selection import train_test_split\nfrom sklearn.ensemble import RandomForestRegressor,GradientBoostingRegressor,AdaBoostRegressor,BaggingRegressor\nfrom sklearn.neighbors import KNeighborsRegressor\nfrom sklearn.tree import DecisionTreeRegressor\nfrom sklearn.linear_model import HuberRegressor,RANSACRegressor,BayesianRidge\nfrom sklearn.svm import SVR","metadata":{"_uuid":"8f2839f25d086af736a60e9eeb907d3b93b6e0e5","_cell_guid":"b1076dfc-b9ad-4769-8c92-a6c4dae69d19","trusted":true,"execution":{"iopub.status.busy":"2024-12-05T12:18:54.954710Z","iopub.execute_input":"2024-12-05T12:18:54.955188Z","iopub.status.idle":"2024-12-05T12:18:54.962107Z","shell.execute_reply.started":"2024-12-05T12:18:54.955142Z","shell.execute_reply":"2024-12-05T12:18:54.960825Z"}},"outputs":[],"execution_count":35},{"cell_type":"code","source":"df=pd.read_csv(\"/kaggle/input/the-evolution-of-genes-in-viruses-and-bacteria/NCBI_Escherichia_coli_genomes.csv\")\ndf","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2024-12-05T11:58:41.478364Z","iopub.execute_input":"2024-12-05T11:58:41.478712Z","iopub.status.idle":"2024-12-05T11:58:41.900975Z","shell.execute_reply.started":"2024-12-05T11:58:41.478674Z","shell.execute_reply":"2024-12-05T11:58:41.899712Z"}},"outputs":[{"execution_count":3,"output_type":"execute_result","data":{"text/plain":"         #Organism Name                              Organism Groups  \\\n0      Escherichia coli  Bacteria;Proteobacteria;Gammaproteobacteria   \n1      Escherichia coli  Bacteria;Proteobacteria;Gammaproteobacteria   \n2      Escherichia coli  Bacteria;Proteobacteria;Gammaproteobacteria   \n3      Escherichia coli  Bacteria;Proteobacteria;Gammaproteobacteria   \n4      Escherichia coli  Bacteria;Proteobacteria;Gammaproteobacteria   \n...                 ...                                          ...   \n24003  Escherichia coli  Bacteria;Proteobacteria;Gammaproteobacteria   \n24004  Escherichia coli  Bacteria;Proteobacteria;Gammaproteobacteria   \n24005  Escherichia coli  Bacteria;Proteobacteria;Gammaproteobacteria   \n24006  Escherichia coli  Bacteria;Proteobacteria;Gammaproteobacteria   \n24007  Escherichia coli  Bacteria;Proteobacteria;Gammaproteobacteria   \n\n             Strain     BioSample   BioProject         Assembly        Level  \\\n0           IMT2125  SAMEA2272029      PRJEB51  GCA_000308975.2     Scaffold   \n1          NCTC9050  SAMEA3376907    PRJEB6403  GCA_900448985.1       Contig   \n2          NCTC9026  SAMEA3368324    PRJEB6403  GCA_900448875.1       Contig   \n3      FDAARGOS_398  SAMN07312442  PRJNA231221  GCA_002554475.1       Contig   \n4      FDAARGOS_399  SAMN07312443  PRJNA231221  GCA_002554675.1       Contig   \n...             ...           ...          ...              ...          ...   \n24003          28.1  SAMN13613893  PRJNA596173  GCA_014170865.1   Chromosome   \n24004         EC931  SAMN14164122  PRJNA608094  GCA_014522225.1     Complete   \n24005      BR29-DEC  SAMN10748603  PRJNA511007  GCA_004116895.1   Chromosome   \n24006  STEC2018-553  SAMN16993384  PRJNA682534  GCA_016403675.2   Chromosome   \n24007  STEC2018-607  SAMN16993387  PRJNA682534  GCA_016403625.2   Chromosome   \n\n       Size(Mb)      GC%                                          Replicons  \\\n0       4.75415  50.7000                                                NaN   \n1       5.84429  50.6000                                                NaN   \n2       5.69529  50.6000                                                NaN   \n3       5.67685  50.4000                                                NaN   \n4       5.63340  50.5000                                                NaN   \n...         ...      ...                                                ...   \n24003   5.10845  50.7001  chromosome:NZ_CP059931.1/CP059931.1; plasmid p...   \n24004   5.13963  50.7500  chromosome:NZ_CP049118.1/CP049118.1; plasmid p...   \n24005   5.01010  50.8489  chromosome:NZ_CP035339.1/CP035339.1; plasmid u...   \n24006   5.69251  50.7547  chromosome:NZ_CP075665.1/CP075665.1; plasmid p...   \n24007   5.62470  50.6071  chromosome:NZ_CP075697.1/CP075697.1; plasmid p...   \n\n          WGS  Scaffolds   CDS          Release Date  \\\n0      CAJT01          1  4266  2012-08-01T00:00:00Z   \n1      UGBX01          4  5283  2018-07-30T00:00:00Z   \n2      UGAL01          2  5246  2018-07-30T00:00:00Z   \n3      PDED01          4  5210  2017-10-16T00:00:00Z   \n4      PDEK01          3  5187  2017-10-16T00:00:00Z   \n...       ...        ...   ...                   ...   \n24003     NaN         13  4579  2020-08-12T00:00:00Z   \n24004     NaN         14  4636  2020-09-10T00:00:00Z   \n24005     NaN         14  4652  2019-01-29T00:00:00Z   \n24006     NaN         15  5319  2021-06-09T00:00:00Z   \n24007     NaN         18  5284  2021-06-09T00:00:00Z   \n\n                                             GenBank FTP  \\\n0      ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...   \n1      ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900...   \n2      ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900...   \n3      ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/002...   \n4      ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/002...   \n...                                                  ...   \n24003  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/014...   \n24004  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/014...   \n24005  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/004...   \n24006  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/016...   \n24007  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/016...   \n\n                                              RefSeq FTP  \n0      ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000...  \n1      ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/900...  \n2      ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/900...  \n3      ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/002...  \n4      ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/002...  \n...                                                  ...  \n24003  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/014...  \n24004  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/014...  \n24005  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/004...  \n24006  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/016...  \n24007  ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/016...  \n\n[24008 rows x 16 columns]","text/html":"<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>#Organism Name</th>\n      <th>Organism Groups</th>\n      <th>Strain</th>\n      <th>BioSample</th>\n      <th>BioProject</th>\n      <th>Assembly</th>\n      <th>Level</th>\n      <th>Size(Mb)</th>\n      <th>GC%</th>\n      <th>Replicons</th>\n      <th>WGS</th>\n      <th>Scaffolds</th>\n      <th>CDS</th>\n      <th>Release Date</th>\n      <th>GenBank FTP</th>\n      <th>RefSeq FTP</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>0</th>\n      <td>Escherichia coli</td>\n      <td>Bacteria;Proteobacteria;Gammaproteobacteria</td>\n      <td>IMT2125</td>\n      <td>SAMEA2272029</td>\n      <td>PRJEB51</td>\n      <td>GCA_000308975.2</td>\n      <td>Scaffold</td>\n      <td>4.75415</td>\n      <td>50.7000</td>\n      <td>NaN</td>\n      <td>CAJT01</td>\n      <td>1</td>\n      <td>4266</td>\n      <td>2012-08-01T00:00:00Z</td>\n      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/000...</td>\n      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000...</td>\n    </tr>\n    <tr>\n      <th>1</th>\n      <td>Escherichia coli</td>\n      <td>Bacteria;Proteobacteria;Gammaproteobacteria</td>\n      <td>NCTC9050</td>\n      <td>SAMEA3376907</td>\n      <td>PRJEB6403</td>\n      <td>GCA_900448985.1</td>\n      <td>Contig</td>\n      <td>5.84429</td>\n      <td>50.6000</td>\n      <td>NaN</td>\n      <td>UGBX01</td>\n      <td>4</td>\n      <td>5283</td>\n      <td>2018-07-30T00:00:00Z</td>\n      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900...</td>\n      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/900...</td>\n    </tr>\n    <tr>\n      <th>2</th>\n      <td>Escherichia coli</td>\n      <td>Bacteria;Proteobacteria;Gammaproteobacteria</td>\n      <td>NCTC9026</td>\n      <td>SAMEA3368324</td>\n      <td>PRJEB6403</td>\n      <td>GCA_900448875.1</td>\n      <td>Contig</td>\n      <td>5.69529</td>\n      <td>50.6000</td>\n      <td>NaN</td>\n      <td>UGAL01</td>\n      <td>2</td>\n      <td>5246</td>\n      <td>2018-07-30T00:00:00Z</td>\n      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/900...</td>\n      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/900...</td>\n    </tr>\n    <tr>\n      <th>3</th>\n      <td>Escherichia coli</td>\n      <td>Bacteria;Proteobacteria;Gammaproteobacteria</td>\n      <td>FDAARGOS_398</td>\n      <td>SAMN07312442</td>\n      <td>PRJNA231221</td>\n      <td>GCA_002554475.1</td>\n      <td>Contig</td>\n      <td>5.67685</td>\n      <td>50.4000</td>\n      <td>NaN</td>\n      <td>PDED01</td>\n      <td>4</td>\n      <td>5210</td>\n      <td>2017-10-16T00:00:00Z</td>\n      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/002...</td>\n      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/002...</td>\n    </tr>\n    <tr>\n      <th>4</th>\n      <td>Escherichia coli</td>\n      <td>Bacteria;Proteobacteria;Gammaproteobacteria</td>\n      <td>FDAARGOS_399</td>\n      <td>SAMN07312443</td>\n      <td>PRJNA231221</td>\n      <td>GCA_002554675.1</td>\n      <td>Contig</td>\n      <td>5.63340</td>\n      <td>50.5000</td>\n      <td>NaN</td>\n      <td>PDEK01</td>\n      <td>3</td>\n      <td>5187</td>\n      <td>2017-10-16T00:00:00Z</td>\n      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/002...</td>\n      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/002...</td>\n    </tr>\n    <tr>\n      <th>...</th>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n    </tr>\n    <tr>\n      <th>24003</th>\n      <td>Escherichia coli</td>\n      <td>Bacteria;Proteobacteria;Gammaproteobacteria</td>\n      <td>28.1</td>\n      <td>SAMN13613893</td>\n      <td>PRJNA596173</td>\n      <td>GCA_014170865.1</td>\n      <td>Chromosome</td>\n      <td>5.10845</td>\n      <td>50.7001</td>\n      <td>chromosome:NZ_CP059931.1/CP059931.1; plasmid p...</td>\n      <td>NaN</td>\n      <td>13</td>\n      <td>4579</td>\n      <td>2020-08-12T00:00:00Z</td>\n      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/014...</td>\n      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/014...</td>\n    </tr>\n    <tr>\n      <th>24004</th>\n      <td>Escherichia coli</td>\n      <td>Bacteria;Proteobacteria;Gammaproteobacteria</td>\n      <td>EC931</td>\n      <td>SAMN14164122</td>\n      <td>PRJNA608094</td>\n      <td>GCA_014522225.1</td>\n      <td>Complete</td>\n      <td>5.13963</td>\n      <td>50.7500</td>\n      <td>chromosome:NZ_CP049118.1/CP049118.1; plasmid p...</td>\n      <td>NaN</td>\n      <td>14</td>\n      <td>4636</td>\n      <td>2020-09-10T00:00:00Z</td>\n      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/014...</td>\n      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/014...</td>\n    </tr>\n    <tr>\n      <th>24005</th>\n      <td>Escherichia coli</td>\n      <td>Bacteria;Proteobacteria;Gammaproteobacteria</td>\n      <td>BR29-DEC</td>\n      <td>SAMN10748603</td>\n      <td>PRJNA511007</td>\n      <td>GCA_004116895.1</td>\n      <td>Chromosome</td>\n      <td>5.01010</td>\n      <td>50.8489</td>\n      <td>chromosome:NZ_CP035339.1/CP035339.1; plasmid u...</td>\n      <td>NaN</td>\n      <td>14</td>\n      <td>4652</td>\n      <td>2019-01-29T00:00:00Z</td>\n      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/004...</td>\n      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/004...</td>\n    </tr>\n    <tr>\n      <th>24006</th>\n      <td>Escherichia coli</td>\n      <td>Bacteria;Proteobacteria;Gammaproteobacteria</td>\n      <td>STEC2018-553</td>\n      <td>SAMN16993384</td>\n      <td>PRJNA682534</td>\n      <td>GCA_016403675.2</td>\n      <td>Chromosome</td>\n      <td>5.69251</td>\n      <td>50.7547</td>\n      <td>chromosome:NZ_CP075665.1/CP075665.1; plasmid p...</td>\n      <td>NaN</td>\n      <td>15</td>\n      <td>5319</td>\n      <td>2021-06-09T00:00:00Z</td>\n      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/016...</td>\n      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/016...</td>\n    </tr>\n    <tr>\n      <th>24007</th>\n      <td>Escherichia coli</td>\n      <td>Bacteria;Proteobacteria;Gammaproteobacteria</td>\n      <td>STEC2018-607</td>\n      <td>SAMN16993387</td>\n      <td>PRJNA682534</td>\n      <td>GCA_016403625.2</td>\n      <td>Chromosome</td>\n      <td>5.62470</td>\n      <td>50.6071</td>\n      <td>chromosome:NZ_CP075697.1/CP075697.1; plasmid p...</td>\n      <td>NaN</td>\n      <td>18</td>\n      <td>5284</td>\n      <td>2021-06-09T00:00:00Z</td>\n      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCA/016...</td>\n      <td>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/016...</td>\n    </tr>\n  </tbody>\n</table>\n<p>24008 rows × 16 columns</p>\n</div>"},"metadata":{}}],"execution_count":3},{"cell_type":"code","source":"df.isnull().sum()","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2024-12-05T11:58:41.902397Z","iopub.execute_input":"2024-12-05T11:58:41.903286Z","iopub.status.idle":"2024-12-05T11:58:41.942947Z","shell.execute_reply.started":"2024-12-05T11:58:41.903249Z","shell.execute_reply":"2024-12-05T11:58:41.941660Z"}},"outputs":[{"execution_count":4,"output_type":"execute_result","data":{"text/plain":"#Organism Name         0\nOrganism Groups        0\nStrain                21\nBioSample              0\nBioProject             0\nAssembly               0\nLevel                  0\nSize(Mb)               0\nGC%                    0\nReplicons          21858\nWGS                 1976\nScaffolds              0\nCDS                    0\nRelease Date           0\nGenBank FTP            0\nRefSeq FTP           408\ndtype: int64"},"metadata":{}}],"execution_count":4},{"cell_type":"code","source":"df=df.drop(columns=[\"GenBank FTP\",\"RefSeq FTP\",\"Replicons\",\"Release Date\"])\ndf=df.dropna(how=\"any\",axis=0)\ndf","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2024-12-05T11:58:41.945789Z","iopub.execute_input":"2024-12-05T11:58:41.946281Z","iopub.status.idle":"2024-12-05T11:58:41.989839Z","shell.execute_reply.started":"2024-12-05T11:58:41.946231Z","shell.execute_reply":"2024-12-05T11:58:41.988711Z"}},"outputs":[{"execution_count":5,"output_type":"execute_result","data":{"text/plain":"         #Organism Name                              Organism Groups  \\\n0      Escherichia coli  Bacteria;Proteobacteria;Gammaproteobacteria   \n1      Escherichia coli  Bacteria;Proteobacteria;Gammaproteobacteria   \n2      Escherichia coli  Bacteria;Proteobacteria;Gammaproteobacteria   \n3      Escherichia coli  Bacteria;Proteobacteria;Gammaproteobacteria   \n4      Escherichia coli  Bacteria;Proteobacteria;Gammaproteobacteria   \n...                 ...                                          ...   \n23916  Escherichia coli  Bacteria;Proteobacteria;Gammaproteobacteria   \n23917  Escherichia coli  Bacteria;Proteobacteria;Gammaproteobacteria   \n23918  Escherichia coli  Bacteria;Proteobacteria;Gammaproteobacteria   \n23977  Escherichia coli  Bacteria;Proteobacteria;Gammaproteobacteria   \n23978  Escherichia coli  Bacteria;Proteobacteria;Gammaproteobacteria   \n\n             Strain     BioSample   BioProject         Assembly     Level  \\\n0           IMT2125  SAMEA2272029      PRJEB51  GCA_000308975.2  Scaffold   \n1          NCTC9050  SAMEA3376907    PRJEB6403  GCA_900448985.1    Contig   \n2          NCTC9026  SAMEA3368324    PRJEB6403  GCA_900448875.1    Contig   \n3      FDAARGOS_398  SAMN07312442  PRJNA231221  GCA_002554475.1    Contig   \n4      FDAARGOS_399  SAMN07312443  PRJNA231221  GCA_002554675.1    Contig   \n...             ...           ...          ...              ...       ...   \n23916       URMC_10  SAMN10620090  PRJNA510429  GCA_004568935.1    Contig   \n23917       URMC_28  SAMN10620105  PRJNA510429  GCA_004567755.1    Contig   \n23918       URMC_39  SAMN10620114  PRJNA510429  GCA_004567695.1    Contig   \n23977       EC-2026  SAMN15580231  PRJNA647258  GCA_016415065.1  Scaffold   \n23978       URMC_57  SAMN10620129  PRJNA510429  GCA_004566955.1    Contig   \n\n       Size(Mb)      GC%       WGS  Scaffolds   CDS  \n0       4.75415  50.7000    CAJT01          1  4266  \n1       5.84429  50.6000    UGBX01          4  5283  \n2       5.69529  50.6000    UGAL01          2  5246  \n3       5.67685  50.4000    PDED01          4  5210  \n4       5.63340  50.5000    PDEK01          3  5187  \n...         ...      ...       ...        ...   ...  \n23916   5.50550  50.8013    RZAC01         86  5195  \n23917   5.11835  50.6198    RZDT01        116  4738  \n23918   5.37304  50.8666    RZED01        166  5027  \n23977   4.92680  50.5316  JACEFM01         76  4480  \n23978   5.20087  50.4404    RZES01        119  4850  \n\n[22011 rows x 12 columns]","text/html":"<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>#Organism Name</th>\n      <th>Organism Groups</th>\n      <th>Strain</th>\n      <th>BioSample</th>\n      <th>BioProject</th>\n      <th>Assembly</th>\n      <th>Level</th>\n      <th>Size(Mb)</th>\n      <th>GC%</th>\n      <th>WGS</th>\n      <th>Scaffolds</th>\n      <th>CDS</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>0</th>\n      <td>Escherichia coli</td>\n      <td>Bacteria;Proteobacteria;Gammaproteobacteria</td>\n      <td>IMT2125</td>\n      <td>SAMEA2272029</td>\n      <td>PRJEB51</td>\n      <td>GCA_000308975.2</td>\n      <td>Scaffold</td>\n      <td>4.75415</td>\n      <td>50.7000</td>\n      <td>CAJT01</td>\n      <td>1</td>\n      <td>4266</td>\n    </tr>\n    <tr>\n      <th>1</th>\n      <td>Escherichia coli</td>\n      <td>Bacteria;Proteobacteria;Gammaproteobacteria</td>\n      <td>NCTC9050</td>\n      <td>SAMEA3376907</td>\n      <td>PRJEB6403</td>\n      <td>GCA_900448985.1</td>\n      <td>Contig</td>\n      <td>5.84429</td>\n      <td>50.6000</td>\n      <td>UGBX01</td>\n      <td>4</td>\n      <td>5283</td>\n    </tr>\n    <tr>\n      <th>2</th>\n      <td>Escherichia coli</td>\n      <td>Bacteria;Proteobacteria;Gammaproteobacteria</td>\n      <td>NCTC9026</td>\n      <td>SAMEA3368324</td>\n      <td>PRJEB6403</td>\n      <td>GCA_900448875.1</td>\n      <td>Contig</td>\n      <td>5.69529</td>\n      <td>50.6000</td>\n      <td>UGAL01</td>\n      <td>2</td>\n      <td>5246</td>\n    </tr>\n    <tr>\n      <th>3</th>\n      <td>Escherichia coli</td>\n      <td>Bacteria;Proteobacteria;Gammaproteobacteria</td>\n      <td>FDAARGOS_398</td>\n      <td>SAMN07312442</td>\n      <td>PRJNA231221</td>\n      <td>GCA_002554475.1</td>\n      <td>Contig</td>\n      <td>5.67685</td>\n      <td>50.4000</td>\n      <td>PDED01</td>\n      <td>4</td>\n      <td>5210</td>\n    </tr>\n    <tr>\n      <th>4</th>\n      <td>Escherichia coli</td>\n      <td>Bacteria;Proteobacteria;Gammaproteobacteria</td>\n      <td>FDAARGOS_399</td>\n      <td>SAMN07312443</td>\n      <td>PRJNA231221</td>\n      <td>GCA_002554675.1</td>\n      <td>Contig</td>\n      <td>5.63340</td>\n      <td>50.5000</td>\n      <td>PDEK01</td>\n      <td>3</td>\n      <td>5187</td>\n    </tr>\n    <tr>\n      <th>...</th>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n    </tr>\n    <tr>\n      <th>23916</th>\n      <td>Escherichia coli</td>\n      <td>Bacteria;Proteobacteria;Gammaproteobacteria</td>\n      <td>URMC_10</td>\n      <td>SAMN10620090</td>\n      <td>PRJNA510429</td>\n      <td>GCA_004568935.1</td>\n      <td>Contig</td>\n      <td>5.50550</td>\n      <td>50.8013</td>\n      <td>RZAC01</td>\n      <td>86</td>\n      <td>5195</td>\n    </tr>\n    <tr>\n      <th>23917</th>\n      <td>Escherichia coli</td>\n      <td>Bacteria;Proteobacteria;Gammaproteobacteria</td>\n      <td>URMC_28</td>\n      <td>SAMN10620105</td>\n      <td>PRJNA510429</td>\n      <td>GCA_004567755.1</td>\n      <td>Contig</td>\n      <td>5.11835</td>\n      <td>50.6198</td>\n      <td>RZDT01</td>\n      <td>116</td>\n      <td>4738</td>\n    </tr>\n    <tr>\n      <th>23918</th>\n      <td>Escherichia coli</td>\n      <td>Bacteria;Proteobacteria;Gammaproteobacteria</td>\n      <td>URMC_39</td>\n      <td>SAMN10620114</td>\n      <td>PRJNA510429</td>\n      <td>GCA_004567695.1</td>\n      <td>Contig</td>\n      <td>5.37304</td>\n      <td>50.8666</td>\n      <td>RZED01</td>\n      <td>166</td>\n      <td>5027</td>\n    </tr>\n    <tr>\n      <th>23977</th>\n      <td>Escherichia coli</td>\n      <td>Bacteria;Proteobacteria;Gammaproteobacteria</td>\n      <td>EC-2026</td>\n      <td>SAMN15580231</td>\n      <td>PRJNA647258</td>\n      <td>GCA_016415065.1</td>\n      <td>Scaffold</td>\n      <td>4.92680</td>\n      <td>50.5316</td>\n      <td>JACEFM01</td>\n      <td>76</td>\n      <td>4480</td>\n    </tr>\n    <tr>\n      <th>23978</th>\n      <td>Escherichia coli</td>\n      <td>Bacteria;Proteobacteria;Gammaproteobacteria</td>\n      <td>URMC_57</td>\n      <td>SAMN10620129</td>\n      <td>PRJNA510429</td>\n      <td>GCA_004566955.1</td>\n      <td>Contig</td>\n      <td>5.20087</td>\n      <td>50.4404</td>\n      <td>RZES01</td>\n      <td>119</td>\n      <td>4850</td>\n    </tr>\n  </tbody>\n</table>\n<p>22011 rows × 12 columns</p>\n</div>"},"metadata":{}}],"execution_count":5},{"cell_type":"code","source":"dd=df[\"Organism Groups\"].str.split(\";\",expand=True)\ndd","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2024-12-05T11:58:41.990975Z","iopub.execute_input":"2024-12-05T11:58:41.991355Z","iopub.status.idle":"2024-12-05T11:58:42.127989Z","shell.execute_reply.started":"2024-12-05T11:58:41.991321Z","shell.execute_reply":"2024-12-05T11:58:42.126504Z"}},"outputs":[{"execution_count":6,"output_type":"execute_result","data":{"text/plain":"              0               1                    2\n0      Bacteria  Proteobacteria  Gammaproteobacteria\n1      Bacteria  Proteobacteria  Gammaproteobacteria\n2      Bacteria  Proteobacteria  Gammaproteobacteria\n3      Bacteria  Proteobacteria  Gammaproteobacteria\n4      Bacteria  Proteobacteria  Gammaproteobacteria\n...         ...             ...                  ...\n23916  Bacteria  Proteobacteria  Gammaproteobacteria\n23917  Bacteria  Proteobacteria  Gammaproteobacteria\n23918  Bacteria  Proteobacteria  Gammaproteobacteria\n23977  Bacteria  Proteobacteria  Gammaproteobacteria\n23978  Bacteria  Proteobacteria  Gammaproteobacteria\n\n[22011 rows x 3 columns]","text/html":"<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>0</th>\n      <th>1</th>\n      <th>2</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>0</th>\n      <td>Bacteria</td>\n      <td>Proteobacteria</td>\n      <td>Gammaproteobacteria</td>\n    </tr>\n    <tr>\n      <th>1</th>\n      <td>Bacteria</td>\n      <td>Proteobacteria</td>\n      <td>Gammaproteobacteria</td>\n    </tr>\n    <tr>\n      <th>2</th>\n      <td>Bacteria</td>\n      <td>Proteobacteria</td>\n      <td>Gammaproteobacteria</td>\n    </tr>\n    <tr>\n      <th>3</th>\n      <td>Bacteria</td>\n      <td>Proteobacteria</td>\n      <td>Gammaproteobacteria</td>\n    </tr>\n    <tr>\n      <th>4</th>\n      <td>Bacteria</td>\n      <td>Proteobacteria</td>\n      <td>Gammaproteobacteria</td>\n    </tr>\n    <tr>\n      <th>...</th>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n    </tr>\n    <tr>\n      <th>23916</th>\n      <td>Bacteria</td>\n      <td>Proteobacteria</td>\n      <td>Gammaproteobacteria</td>\n    </tr>\n    <tr>\n      <th>23917</th>\n      <td>Bacteria</td>\n      <td>Proteobacteria</td>\n      <td>Gammaproteobacteria</td>\n    </tr>\n    <tr>\n      <th>23918</th>\n      <td>Bacteria</td>\n      <td>Proteobacteria</td>\n      <td>Gammaproteobacteria</td>\n    </tr>\n    <tr>\n      <th>23977</th>\n      <td>Bacteria</td>\n      <td>Proteobacteria</td>\n      <td>Gammaproteobacteria</td>\n    </tr>\n    <tr>\n      <th>23978</th>\n      <td>Bacteria</td>\n      <td>Proteobacteria</td>\n      <td>Gammaproteobacteria</td>\n    </tr>\n  </tbody>\n</table>\n<p>22011 rows × 3 columns</p>\n</div>"},"metadata":{}}],"execution_count":6},{"cell_type":"code","source":"concaty=pd.concat([df,dd],axis=1).drop(columns=[\"Organism Groups\"])\nconcaty","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2024-12-05T11:58:42.129678Z","iopub.execute_input":"2024-12-05T11:58:42.130199Z","iopub.status.idle":"2024-12-05T11:58:42.168600Z","shell.execute_reply.started":"2024-12-05T11:58:42.130134Z","shell.execute_reply":"2024-12-05T11:58:42.167388Z"}},"outputs":[{"execution_count":7,"output_type":"execute_result","data":{"text/plain":"         #Organism Name        Strain     BioSample   BioProject  \\\n0      Escherichia coli       IMT2125  SAMEA2272029      PRJEB51   \n1      Escherichia coli      NCTC9050  SAMEA3376907    PRJEB6403   \n2      Escherichia coli      NCTC9026  SAMEA3368324    PRJEB6403   \n3      Escherichia coli  FDAARGOS_398  SAMN07312442  PRJNA231221   \n4      Escherichia coli  FDAARGOS_399  SAMN07312443  PRJNA231221   \n...                 ...           ...           ...          ...   \n23916  Escherichia coli       URMC_10  SAMN10620090  PRJNA510429   \n23917  Escherichia coli       URMC_28  SAMN10620105  PRJNA510429   \n23918  Escherichia coli       URMC_39  SAMN10620114  PRJNA510429   \n23977  Escherichia coli       EC-2026  SAMN15580231  PRJNA647258   \n23978  Escherichia coli       URMC_57  SAMN10620129  PRJNA510429   \n\n              Assembly     Level  Size(Mb)      GC%       WGS  Scaffolds  \\\n0      GCA_000308975.2  Scaffold   4.75415  50.7000    CAJT01          1   \n1      GCA_900448985.1    Contig   5.84429  50.6000    UGBX01          4   \n2      GCA_900448875.1    Contig   5.69529  50.6000    UGAL01          2   \n3      GCA_002554475.1    Contig   5.67685  50.4000    PDED01          4   \n4      GCA_002554675.1    Contig   5.63340  50.5000    PDEK01          3   \n...                ...       ...       ...      ...       ...        ...   \n23916  GCA_004568935.1    Contig   5.50550  50.8013    RZAC01         86   \n23917  GCA_004567755.1    Contig   5.11835  50.6198    RZDT01        116   \n23918  GCA_004567695.1    Contig   5.37304  50.8666    RZED01        166   \n23977  GCA_016415065.1  Scaffold   4.92680  50.5316  JACEFM01         76   \n23978  GCA_004566955.1    Contig   5.20087  50.4404    RZES01        119   \n\n        CDS         0               1                    2  \n0      4266  Bacteria  Proteobacteria  Gammaproteobacteria  \n1      5283  Bacteria  Proteobacteria  Gammaproteobacteria  \n2      5246  Bacteria  Proteobacteria  Gammaproteobacteria  \n3      5210  Bacteria  Proteobacteria  Gammaproteobacteria  \n4      5187  Bacteria  Proteobacteria  Gammaproteobacteria  \n...     ...       ...             ...                  ...  \n23916  5195  Bacteria  Proteobacteria  Gammaproteobacteria  \n23917  4738  Bacteria  Proteobacteria  Gammaproteobacteria  \n23918  5027  Bacteria  Proteobacteria  Gammaproteobacteria  \n23977  4480  Bacteria  Proteobacteria  Gammaproteobacteria  \n23978  4850  Bacteria  Proteobacteria  Gammaproteobacteria  \n\n[22011 rows x 14 columns]","text/html":"<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>#Organism Name</th>\n      <th>Strain</th>\n      <th>BioSample</th>\n      <th>BioProject</th>\n      <th>Assembly</th>\n      <th>Level</th>\n      <th>Size(Mb)</th>\n      <th>GC%</th>\n      <th>WGS</th>\n      <th>Scaffolds</th>\n      <th>CDS</th>\n      <th>0</th>\n      <th>1</th>\n      <th>2</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>0</th>\n      <td>Escherichia coli</td>\n      <td>IMT2125</td>\n      <td>SAMEA2272029</td>\n      <td>PRJEB51</td>\n      <td>GCA_000308975.2</td>\n      <td>Scaffold</td>\n      <td>4.75415</td>\n      <td>50.7000</td>\n      <td>CAJT01</td>\n      <td>1</td>\n      <td>4266</td>\n      <td>Bacteria</td>\n      <td>Proteobacteria</td>\n      <td>Gammaproteobacteria</td>\n    </tr>\n    <tr>\n      <th>1</th>\n      <td>Escherichia coli</td>\n      <td>NCTC9050</td>\n      <td>SAMEA3376907</td>\n      <td>PRJEB6403</td>\n      <td>GCA_900448985.1</td>\n      <td>Contig</td>\n      <td>5.84429</td>\n      <td>50.6000</td>\n      <td>UGBX01</td>\n      <td>4</td>\n      <td>5283</td>\n      <td>Bacteria</td>\n      <td>Proteobacteria</td>\n      <td>Gammaproteobacteria</td>\n    </tr>\n    <tr>\n      <th>2</th>\n      <td>Escherichia coli</td>\n      <td>NCTC9026</td>\n      <td>SAMEA3368324</td>\n      <td>PRJEB6403</td>\n      <td>GCA_900448875.1</td>\n      <td>Contig</td>\n      <td>5.69529</td>\n      <td>50.6000</td>\n      <td>UGAL01</td>\n      <td>2</td>\n      <td>5246</td>\n      <td>Bacteria</td>\n      <td>Proteobacteria</td>\n      <td>Gammaproteobacteria</td>\n    </tr>\n    <tr>\n      <th>3</th>\n      <td>Escherichia coli</td>\n      <td>FDAARGOS_398</td>\n      <td>SAMN07312442</td>\n      <td>PRJNA231221</td>\n      <td>GCA_002554475.1</td>\n      <td>Contig</td>\n      <td>5.67685</td>\n      <td>50.4000</td>\n      <td>PDED01</td>\n      <td>4</td>\n      <td>5210</td>\n      <td>Bacteria</td>\n      <td>Proteobacteria</td>\n      <td>Gammaproteobacteria</td>\n    </tr>\n    <tr>\n      <th>4</th>\n      <td>Escherichia coli</td>\n      <td>FDAARGOS_399</td>\n      <td>SAMN07312443</td>\n      <td>PRJNA231221</td>\n      <td>GCA_002554675.1</td>\n      <td>Contig</td>\n      <td>5.63340</td>\n      <td>50.5000</td>\n      <td>PDEK01</td>\n      <td>3</td>\n      <td>5187</td>\n      <td>Bacteria</td>\n      <td>Proteobacteria</td>\n      <td>Gammaproteobacteria</td>\n    </tr>\n    <tr>\n      <th>...</th>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n    </tr>\n    <tr>\n      <th>23916</th>\n      <td>Escherichia coli</td>\n      <td>URMC_10</td>\n      <td>SAMN10620090</td>\n      <td>PRJNA510429</td>\n      <td>GCA_004568935.1</td>\n      <td>Contig</td>\n      <td>5.50550</td>\n      <td>50.8013</td>\n      <td>RZAC01</td>\n      <td>86</td>\n      <td>5195</td>\n      <td>Bacteria</td>\n      <td>Proteobacteria</td>\n      <td>Gammaproteobacteria</td>\n    </tr>\n    <tr>\n      <th>23917</th>\n      <td>Escherichia coli</td>\n      <td>URMC_28</td>\n      <td>SAMN10620105</td>\n      <td>PRJNA510429</td>\n      <td>GCA_004567755.1</td>\n      <td>Contig</td>\n      <td>5.11835</td>\n      <td>50.6198</td>\n      <td>RZDT01</td>\n      <td>116</td>\n      <td>4738</td>\n      <td>Bacteria</td>\n      <td>Proteobacteria</td>\n      <td>Gammaproteobacteria</td>\n    </tr>\n    <tr>\n      <th>23918</th>\n      <td>Escherichia coli</td>\n      <td>URMC_39</td>\n      <td>SAMN10620114</td>\n      <td>PRJNA510429</td>\n      <td>GCA_004567695.1</td>\n      <td>Contig</td>\n      <td>5.37304</td>\n      <td>50.8666</td>\n      <td>RZED01</td>\n      <td>166</td>\n      <td>5027</td>\n      <td>Bacteria</td>\n      <td>Proteobacteria</td>\n      <td>Gammaproteobacteria</td>\n    </tr>\n    <tr>\n      <th>23977</th>\n      <td>Escherichia coli</td>\n      <td>EC-2026</td>\n      <td>SAMN15580231</td>\n      <td>PRJNA647258</td>\n      <td>GCA_016415065.1</td>\n      <td>Scaffold</td>\n      <td>4.92680</td>\n      <td>50.5316</td>\n      <td>JACEFM01</td>\n      <td>76</td>\n      <td>4480</td>\n      <td>Bacteria</td>\n      <td>Proteobacteria</td>\n      <td>Gammaproteobacteria</td>\n    </tr>\n    <tr>\n      <th>23978</th>\n      <td>Escherichia coli</td>\n      <td>URMC_57</td>\n      <td>SAMN10620129</td>\n      <td>PRJNA510429</td>\n      <td>GCA_004566955.1</td>\n      <td>Contig</td>\n      <td>5.20087</td>\n      <td>50.4404</td>\n      <td>RZES01</td>\n      <td>119</td>\n      <td>4850</td>\n      <td>Bacteria</td>\n      <td>Proteobacteria</td>\n      <td>Gammaproteobacteria</td>\n    </tr>\n  </tbody>\n</table>\n<p>22011 rows × 14 columns</p>\n</div>"},"metadata":{}}],"execution_count":7},{"cell_type":"code","source":"concaty.nunique()","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2024-12-05T11:58:42.170267Z","iopub.execute_input":"2024-12-05T11:58:42.170598Z","iopub.status.idle":"2024-12-05T11:58:42.219614Z","shell.execute_reply.started":"2024-12-05T11:58:42.170567Z","shell.execute_reply":"2024-12-05T11:58:42.218484Z"}},"outputs":[{"execution_count":8,"output_type":"execute_result","data":{"text/plain":"#Organism Name     2244\nStrain            21455\nBioSample         21992\nBioProject         3702\nAssembly          22011\nLevel                 3\nSize(Mb)          19488\nGC%                 273\nWGS               22011\nScaffolds          1127\nCDS                1789\n0                     1\n1                     1\n2                     1\ndtype: int64"},"metadata":{}}],"execution_count":8},{"cell_type":"code","source":"concaty[\"Strain\"]","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2024-12-05T11:58:42.220795Z","iopub.execute_input":"2024-12-05T11:58:42.221114Z","iopub.status.idle":"2024-12-05T11:58:42.229534Z","shell.execute_reply.started":"2024-12-05T11:58:42.221056Z","shell.execute_reply":"2024-12-05T11:58:42.228367Z"}},"outputs":[{"execution_count":9,"output_type":"execute_result","data":{"text/plain":"0             IMT2125\n1            NCTC9050\n2            NCTC9026\n3        FDAARGOS_398\n4        FDAARGOS_399\n             ...     \n23916         URMC_10\n23917         URMC_28\n23918         URMC_39\n23977         EC-2026\n23978         URMC_57\nName: Strain, Length: 22011, dtype: object"},"metadata":{}}],"execution_count":9},{"cell_type":"code","source":"label_encoder=LabelEncoder()\nfor col in concaty.columns:\n    concaty[col]=label_encoder.fit_transform(concaty[col])","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2024-12-05T11:58:54.653659Z","iopub.execute_input":"2024-12-05T11:58:54.654073Z","iopub.status.idle":"2024-12-05T11:58:54.905258Z","shell.execute_reply.started":"2024-12-05T11:58:54.654036Z","shell.execute_reply":"2024-12-05T11:58:54.903814Z"}},"outputs":[],"execution_count":11},{"cell_type":"code","source":"concaty[\"Bacteria\"]=concaty[0]\nconcaty[\"proteoBacteria\"]=concaty[1]\nconcaty[\"Gammaproteobacteria\"]=concaty[2]\nconcaty=concaty.drop(columns=[0,1,2])","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2024-12-05T12:02:07.163522Z","iopub.execute_input":"2024-12-05T12:02:07.163919Z","iopub.status.idle":"2024-12-05T12:02:07.174866Z","shell.execute_reply.started":"2024-12-05T12:02:07.163882Z","shell.execute_reply":"2024-12-05T12:02:07.173584Z"}},"outputs":[],"execution_count":18},{"cell_type":"code","source":"concaty","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2024-12-05T12:02:07.843633Z","iopub.execute_input":"2024-12-05T12:02:07.844278Z","iopub.status.idle":"2024-12-05T12:02:07.863473Z","shell.execute_reply.started":"2024-12-05T12:02:07.844236Z","shell.execute_reply":"2024-12-05T12:02:07.862115Z"}},"outputs":[{"execution_count":19,"output_type":"execute_result","data":{"text/plain":"       #Organism Name  Strain  BioSample  BioProject  Assembly  Level  \\\n0                   0   10798       2435         205       428      2   \n1                   0   15729       2602         206     20689      1   \n2                   0   15714       2589         206     20679      1   \n3                   0    8753      12554         981      8504      1   \n4                   0    8754      12555         981      8506      1   \n...               ...     ...        ...         ...       ...    ...   \n23916               0   19678      16192        2597     12645      1   \n23917               0   19772      16206        2597     12586      1   \n23918               0   19782      16215        2597     12582      1   \n23977               0    6916      20188        3034     18820      2   \n23978               0   19797      16229        2597     12545      1   \n\n       Size(Mb)  GC%    WGS  Scaffolds   CDS  Bacteria  proteoBacteria  \\\n0          1773  166   4053          0   286         0               0   \n1         19290  120  20463          3  1302         0               0   \n2         19038  120  20425          1  1266         0               0   \n3         18993   56  15294          3  1230         0               0   \n4         18835   86  15296          2  1207         0               0   \n...         ...  ...    ...        ...   ...       ...             ...   \n23916     18003  209  18311         85  1215         0               0   \n23917      9616  132  18405        115   758         0               0   \n23918     16046  234  18415        165  1047         0               0   \n23977      4703   95   5975         75   500         0               0   \n23978     11928   62  18430        118   870         0               0   \n\n       Gammaproteobacteria  \n0                        0  \n1                        0  \n2                        0  \n3                        0  \n4                        0  \n...                    ...  \n23916                    0  \n23917                    0  \n23918                    0  \n23977                    0  \n23978                    0  \n\n[22011 rows x 14 columns]","text/html":"<div>\n<style scoped>\n    .dataframe tbody tr th:only-of-type {\n        vertical-align: middle;\n    }\n\n    .dataframe tbody tr th {\n        vertical-align: top;\n    }\n\n    .dataframe thead th {\n        text-align: right;\n    }\n</style>\n<table border=\"1\" class=\"dataframe\">\n  <thead>\n    <tr style=\"text-align: right;\">\n      <th></th>\n      <th>#Organism Name</th>\n      <th>Strain</th>\n      <th>BioSample</th>\n      <th>BioProject</th>\n      <th>Assembly</th>\n      <th>Level</th>\n      <th>Size(Mb)</th>\n      <th>GC%</th>\n      <th>WGS</th>\n      <th>Scaffolds</th>\n      <th>CDS</th>\n      <th>Bacteria</th>\n      <th>proteoBacteria</th>\n      <th>Gammaproteobacteria</th>\n    </tr>\n  </thead>\n  <tbody>\n    <tr>\n      <th>0</th>\n      <td>0</td>\n      <td>10798</td>\n      <td>2435</td>\n      <td>205</td>\n      <td>428</td>\n      <td>2</td>\n      <td>1773</td>\n      <td>166</td>\n      <td>4053</td>\n      <td>0</td>\n      <td>286</td>\n      <td>0</td>\n      <td>0</td>\n      <td>0</td>\n    </tr>\n    <tr>\n      <th>1</th>\n      <td>0</td>\n      <td>15729</td>\n      <td>2602</td>\n      <td>206</td>\n      <td>20689</td>\n      <td>1</td>\n      <td>19290</td>\n      <td>120</td>\n      <td>20463</td>\n      <td>3</td>\n      <td>1302</td>\n      <td>0</td>\n      <td>0</td>\n      <td>0</td>\n    </tr>\n    <tr>\n      <th>2</th>\n      <td>0</td>\n      <td>15714</td>\n      <td>2589</td>\n      <td>206</td>\n      <td>20679</td>\n      <td>1</td>\n      <td>19038</td>\n      <td>120</td>\n      <td>20425</td>\n      <td>1</td>\n      <td>1266</td>\n      <td>0</td>\n      <td>0</td>\n      <td>0</td>\n    </tr>\n    <tr>\n      <th>3</th>\n      <td>0</td>\n      <td>8753</td>\n      <td>12554</td>\n      <td>981</td>\n      <td>8504</td>\n      <td>1</td>\n      <td>18993</td>\n      <td>56</td>\n      <td>15294</td>\n      <td>3</td>\n      <td>1230</td>\n      <td>0</td>\n      <td>0</td>\n      <td>0</td>\n    </tr>\n    <tr>\n      <th>4</th>\n      <td>0</td>\n      <td>8754</td>\n      <td>12555</td>\n      <td>981</td>\n      <td>8506</td>\n      <td>1</td>\n      <td>18835</td>\n      <td>86</td>\n      <td>15296</td>\n      <td>2</td>\n      <td>1207</td>\n      <td>0</td>\n      <td>0</td>\n      <td>0</td>\n    </tr>\n    <tr>\n      <th>...</th>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n      <td>...</td>\n    </tr>\n    <tr>\n      <th>23916</th>\n      <td>0</td>\n      <td>19678</td>\n      <td>16192</td>\n      <td>2597</td>\n      <td>12645</td>\n      <td>1</td>\n      <td>18003</td>\n      <td>209</td>\n      <td>18311</td>\n      <td>85</td>\n      <td>1215</td>\n      <td>0</td>\n      <td>0</td>\n      <td>0</td>\n    </tr>\n    <tr>\n      <th>23917</th>\n      <td>0</td>\n      <td>19772</td>\n      <td>16206</td>\n      <td>2597</td>\n      <td>12586</td>\n      <td>1</td>\n      <td>9616</td>\n      <td>132</td>\n      <td>18405</td>\n      <td>115</td>\n      <td>758</td>\n      <td>0</td>\n      <td>0</td>\n      <td>0</td>\n    </tr>\n    <tr>\n      <th>23918</th>\n      <td>0</td>\n      <td>19782</td>\n      <td>16215</td>\n      <td>2597</td>\n      <td>12582</td>\n      <td>1</td>\n      <td>16046</td>\n      <td>234</td>\n      <td>18415</td>\n      <td>165</td>\n      <td>1047</td>\n      <td>0</td>\n      <td>0</td>\n      <td>0</td>\n    </tr>\n    <tr>\n      <th>23977</th>\n      <td>0</td>\n      <td>6916</td>\n      <td>20188</td>\n      <td>3034</td>\n      <td>18820</td>\n      <td>2</td>\n      <td>4703</td>\n      <td>95</td>\n      <td>5975</td>\n      <td>75</td>\n      <td>500</td>\n      <td>0</td>\n      <td>0</td>\n      <td>0</td>\n    </tr>\n    <tr>\n      <th>23978</th>\n      <td>0</td>\n      <td>19797</td>\n      <td>16229</td>\n      <td>2597</td>\n      <td>12545</td>\n      <td>1</td>\n      <td>11928</td>\n      <td>62</td>\n      <td>18430</td>\n      <td>118</td>\n      <td>870</td>\n      <td>0</td>\n      <td>0</td>\n      <td>0</td>\n    </tr>\n  </tbody>\n</table>\n<p>22011 rows × 14 columns</p>\n</div>"},"metadata":{}}],"execution_count":19},{"cell_type":"code","source":"X=concaty.drop(columns=[\"Strain\"])\nY=concaty[\"Strain\"]","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2024-12-05T12:02:19.268830Z","iopub.execute_input":"2024-12-05T12:02:19.269245Z","iopub.status.idle":"2024-12-05T12:02:19.277977Z","shell.execute_reply.started":"2024-12-05T12:02:19.269208Z","shell.execute_reply":"2024-12-05T12:02:19.276757Z"}},"outputs":[],"execution_count":21},{"cell_type":"code","source":"X_train,X_test,Y_train,Y_test=train_test_split(X,Y,test_size=0.2,random_state=42)","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2024-12-05T12:03:47.103991Z","iopub.execute_input":"2024-12-05T12:03:47.104439Z","iopub.status.idle":"2024-12-05T12:03:47.121514Z","shell.execute_reply.started":"2024-12-05T12:03:47.104401Z","shell.execute_reply":"2024-12-05T12:03:47.120058Z"}},"outputs":[],"execution_count":24},{"cell_type":"code","source":"for i in [RandomForestRegressor(),GradientBoostingRegressor(),AdaBoostRegressor(),BaggingRegressor(),BayesianRidge(),KNeighborsRegressor(),DecisionTreeRegressor(),SVR(),HuberRegressor(),RANSACRegressor()]:\n    random_reg=i\n    random_reg.fit(X_train,Y_train)\n    print(f\"the accuracy of {i}\",random_reg.score(X_test,Y_test))","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2024-12-05T12:19:15.804563Z","iopub.execute_input":"2024-12-05T12:19:15.805101Z","iopub.status.idle":"2024-12-05T12:19:59.826758Z","shell.execute_reply.started":"2024-12-05T12:19:15.805041Z","shell.execute_reply":"2024-12-05T12:19:59.823733Z"}},"outputs":[{"name":"stdout","text":"the accuracy of RandomForestRegressor() 0.8998930687766277\nthe accuracy of GradientBoostingRegressor() 0.5247101288121743\nthe accuracy of AdaBoostRegressor() 0.16262523534728623\nthe accuracy of BaggingRegressor() 0.888170090727004\nthe accuracy of BayesianRidge() 0.03716504713390367\nthe accuracy of KNeighborsRegressor() 0.5443587096189677\nthe accuracy of DecisionTreeRegressor() 0.8228385686289907\nthe accuracy of SVR() 0.010341564125501246\n","output_type":"stream"},{"name":"stderr","text":"/opt/conda/lib/python3.10/site-packages/sklearn/linear_model/_huber.py:342: ConvergenceWarning: lbfgs failed to converge (status=1):\nSTOP: TOTAL NO. of ITERATIONS REACHED LIMIT.\n\nIncrease the number of iterations (max_iter) or scale the data as shown in:\n    https://scikit-learn.org/stable/modules/preprocessing.html\n  self.n_iter_ = _check_optimize_result(\"lbfgs\", opt_res, self.max_iter)\n","output_type":"stream"},{"name":"stdout","text":"the accuracy of HuberRegressor() -0.14317404374841658\nthe accuracy of RANSACRegressor() -0.9589735746180534\n","output_type":"stream"}],"execution_count":36},{"cell_type":"code","source":"X_train.shape","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2024-12-05T12:26:45.358991Z","iopub.execute_input":"2024-12-05T12:26:45.359455Z","iopub.status.idle":"2024-12-05T12:26:45.367257Z","shell.execute_reply.started":"2024-12-05T12:26:45.359402Z","shell.execute_reply":"2024-12-05T12:26:45.365938Z"}},"outputs":[{"execution_count":41,"output_type":"execute_result","data":{"text/plain":"(17608, 13)"},"metadata":{}}],"execution_count":41},{"cell_type":"markdown","source":"# ##Deep Learning  ","metadata":{}},{"cell_type":"code","source":"from keras.layers import Layer,Dense,Flatten,BatchNormalization,Input,LeakyReLU\nfrom keras.initializers import HeNormal,GlorotUniform\nfrom keras.regularizers import l2,l1_l2\nfrom keras.callbacks import EarlyStopping\nfrom keras.models import Model,Sequential\nfrom keras.optimizers import Adam","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2024-12-05T12:30:53.564457Z","iopub.execute_input":"2024-12-05T12:30:53.565579Z","iopub.status.idle":"2024-12-05T12:30:53.572264Z","shell.execute_reply.started":"2024-12-05T12:30:53.565534Z","shell.execute_reply":"2024-12-05T12:30:53.570834Z"}},"outputs":[],"execution_count":50},{"cell_type":"code","source":"inputs=Input(shape=(13,))\nd=Dense(32,activation=\"relu\",kernel_initializer=HeNormal(),kernel_regularizer=l2(0.01))(inputs)\nd=Dense(64,kernel_initializer=HeNormal(),kernel_regularizer=l2(0.01))(d)\nd=LeakyReLU(0.08)(d)\nd=Dense(128,kernel_initializer=GlorotUniform(),kernel_regularizer=l2(0.01))(d)\nd=LeakyReLU(0.08)(d)\nd=Dense(228,kernel_initializer=GlorotUniform(),kernel_regularizer=l2(0.01))(d)\nd=LeakyReLU(0.08)(d)\nd=Dense(528,kernel_initializer=GlorotUniform(),kernel_regularizer=l2(0.01))(d)\nd=LeakyReLU(0.08)(d)\nd=Dense(1028,kernel_initializer=GlorotUniform(),kernel_regularizer=l2(0.01))(d)\nd=LeakyReLU(0.08)(d)\noutputs=Dense(1)(d)\nmodel=Model(inputs,outputs)\noptimizer=Adam(learning_rate=0.001,beta_1=0.99,beta_2=0.99)\nmodel.compile(optimizer=optimizer,loss=\"mse\",metrics=[\"mae\"])\nmodel.fit(X_train,Y_train,epochs=15,validation_data=(X_test,Y_test))","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2024-12-05T12:33:10.135199Z","iopub.execute_input":"2024-12-05T12:33:10.135580Z","iopub.status.idle":"2024-12-05T12:34:35.325776Z","shell.execute_reply.started":"2024-12-05T12:33:10.135548Z","shell.execute_reply":"2024-12-05T12:34:35.324626Z"}},"outputs":[{"name":"stdout","text":"Epoch 1/15\n\u001b[1m551/551\u001b[0m \u001b[32m━━━━━━━━━━━━━━━━━━━━\u001b[0m\u001b[37m\u001b[0m \u001b[1m8s\u001b[0m 10ms/step - loss: 45068904.0000 - mae: 5458.9458 - val_loss: 36802240.0000 - val_mae: 5018.4365\nEpoch 2/15\n\u001b[1m551/551\u001b[0m \u001b[32m━━━━━━━━━━━━━━━━━━━━\u001b[0m\u001b[37m\u001b[0m \u001b[1m5s\u001b[0m 10ms/step - loss: 36138756.0000 - mae: 4857.8213 - val_loss: 35462012.0000 - val_mae: 4641.0176\nEpoch 3/15\n\u001b[1m551/551\u001b[0m \u001b[32m━━━━━━━━━━━━━━━━━━━━\u001b[0m\u001b[37m\u001b[0m \u001b[1m5s\u001b[0m 9ms/step - loss: 33814580.0000 - mae: 4652.0449 - val_loss: 33654104.0000 - val_mae: 4575.7280\nEpoch 4/15\n\u001b[1m551/551\u001b[0m \u001b[32m━━━━━━━━━━━━━━━━━━━━\u001b[0m\u001b[37m\u001b[0m \u001b[1m6s\u001b[0m 11ms/step - loss: 33677888.0000 - mae: 4654.1489 - val_loss: 32555812.0000 - val_mae: 4442.6440\nEpoch 5/15\n\u001b[1m551/551\u001b[0m \u001b[32m━━━━━━━━━━━━━━━━━━━━\u001b[0m\u001b[37m\u001b[0m \u001b[1m6s\u001b[0m 10ms/step - loss: 32970176.0000 - mae: 4578.1592 - val_loss: 32118080.0000 - val_mae: 4512.3145\nEpoch 6/15\n\u001b[1m551/551\u001b[0m \u001b[32m━━━━━━━━━━━━━━━━━━━━\u001b[0m\u001b[37m\u001b[0m \u001b[1m5s\u001b[0m 10ms/step - loss: 31493734.0000 - mae: 4471.7930 - val_loss: 35621116.0000 - val_mae: 4865.1372\nEpoch 7/15\n\u001b[1m551/551\u001b[0m \u001b[32m━━━━━━━━━━━━━━━━━━━━\u001b[0m\u001b[37m\u001b[0m \u001b[1m5s\u001b[0m 10ms/step - loss: 32504366.0000 - mae: 4536.8833 - val_loss: 31996936.0000 - val_mae: 4470.5015\nEpoch 8/15\n\u001b[1m551/551\u001b[0m \u001b[32m━━━━━━━━━━━━━━━━━━━━\u001b[0m\u001b[37m\u001b[0m \u001b[1m5s\u001b[0m 10ms/step - loss: 31525426.0000 - mae: 4426.5542 - val_loss: 31815856.0000 - val_mae: 4427.5386\nEpoch 9/15\n\u001b[1m551/551\u001b[0m \u001b[32m━━━━━━━━━━━━━━━━━━━━\u001b[0m\u001b[37m\u001b[0m \u001b[1m6s\u001b[0m 10ms/step - loss: 30339904.0000 - mae: 4324.2529 - val_loss: 30005222.0000 - val_mae: 4283.1099\nEpoch 10/15\n\u001b[1m551/551\u001b[0m \u001b[32m━━━━━━━━━━━━━━━━━━━━\u001b[0m\u001b[37m\u001b[0m \u001b[1m6s\u001b[0m 10ms/step - loss: 30254002.0000 - mae: 4335.0386 - val_loss: 30192310.0000 - val_mae: 4224.8354\nEpoch 11/15\n\u001b[1m551/551\u001b[0m \u001b[32m━━━━━━━━━━━━━━━━━━━━\u001b[0m\u001b[37m\u001b[0m \u001b[1m5s\u001b[0m 10ms/step - loss: 29813914.0000 - mae: 4275.1157 - val_loss: 31065136.0000 - val_mae: 4455.4263\nEpoch 12/15\n\u001b[1m551/551\u001b[0m \u001b[32m━━━━━━━━━━━━━━━━━━━━\u001b[0m\u001b[37m\u001b[0m \u001b[1m5s\u001b[0m 10ms/step - loss: 29421682.0000 - mae: 4259.6372 - val_loss: 31148772.0000 - val_mae: 4390.1533\nEpoch 13/15\n\u001b[1m551/551\u001b[0m \u001b[32m━━━━━━━━━━━━━━━━━━━━\u001b[0m\u001b[37m\u001b[0m \u001b[1m5s\u001b[0m 10ms/step - loss: 29927526.0000 - mae: 4294.5996 - val_loss: 32061206.0000 - val_mae: 4447.2178\nEpoch 14/15\n\u001b[1m551/551\u001b[0m \u001b[32m━━━━━━━━━━━━━━━━━━━━\u001b[0m\u001b[37m\u001b[0m \u001b[1m5s\u001b[0m 10ms/step - loss: 28516436.0000 - mae: 4165.7529 - val_loss: 28480602.0000 - val_mae: 4146.7666\nEpoch 15/15\n\u001b[1m551/551\u001b[0m \u001b[32m━━━━━━━━━━━━━━━━━━━━\u001b[0m\u001b[37m\u001b[0m \u001b[1m6s\u001b[0m 10ms/step - loss: 27852782.0000 - mae: 4108.3027 - val_loss: 28293678.0000 - val_mae: 4092.6455\n","output_type":"stream"},{"execution_count":54,"output_type":"execute_result","data":{"text/plain":"<keras.src.callbacks.history.History at 0x7a1dc9699090>"},"metadata":{}}],"execution_count":54},{"cell_type":"code","source":"loss,accuracy=model.evaluate(X_test,Y_test)","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2024-12-05T12:35:51.895676Z","iopub.execute_input":"2024-12-05T12:35:51.896116Z","iopub.status.idle":"2024-12-05T12:35:52.533690Z","shell.execute_reply.started":"2024-12-05T12:35:51.896052Z","shell.execute_reply":"2024-12-05T12:35:52.532462Z"}},"outputs":[{"name":"stdout","text":"\u001b[1m138/138\u001b[0m \u001b[32m━━━━━━━━━━━━━━━━━━━━\u001b[0m\u001b[37m\u001b[0m \u001b[1m1s\u001b[0m 4ms/step - loss: 29163294.0000 - mae: 4166.1680\n","output_type":"stream"}],"execution_count":56},{"cell_type":"code","source":"model.predict(X_test.iloc[:10,:])","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2024-12-05T15:11:26.589647Z","iopub.execute_input":"2024-12-05T15:11:26.590075Z","iopub.status.idle":"2024-12-05T15:11:26.750263Z","shell.execute_reply.started":"2024-12-05T15:11:26.590040Z","shell.execute_reply":"2024-12-05T15:11:26.749020Z"}},"outputs":[{"name":"stdout","text":"\u001b[1m1/1\u001b[0m \u001b[32m━━━━━━━━━━━━━━━━━━━━\u001b[0m\u001b[37m\u001b[0m \u001b[1m0s\u001b[0m 96ms/step\n","output_type":"stream"},{"execution_count":78,"output_type":"execute_result","data":{"text/plain":"array([[ 8396.616 ],\n       [12889.564 ],\n       [ 8785.204 ],\n       [13313.317 ],\n       [ 9024.17  ],\n       [ 6688.896 ],\n       [ 7676.043 ],\n       [13160.272 ],\n       [ 7540.0674],\n       [18925.217 ]], dtype=float32)"},"metadata":{}}],"execution_count":78},{"cell_type":"code","source":"Y_test[:10]","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2024-12-05T15:11:50.550129Z","iopub.execute_input":"2024-12-05T15:11:50.550535Z","iopub.status.idle":"2024-12-05T15:11:50.559254Z","shell.execute_reply.started":"2024-12-05T15:11:50.550499Z","shell.execute_reply":"2024-12-05T15:11:50.558118Z"}},"outputs":[{"execution_count":79,"output_type":"execute_result","data":{"text/plain":"11429     9030\n14802    20627\n8323     21046\n16757    17984\n12685     4653\n18783    19110\n3827      9463\n221      17648\n10563     7426\n10492    13694\nName: Strain, dtype: int64"},"metadata":{}}],"execution_count":79},{"cell_type":"markdown","source":"# ##  Quantum Deep Learning","metadata":{}},{"cell_type":"code","source":"!pip install cirq","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2024-12-05T12:39:45.593910Z","iopub.execute_input":"2024-12-05T12:39:45.594382Z","iopub.status.idle":"2024-12-05T12:40:11.454206Z","shell.execute_reply.started":"2024-12-05T12:39:45.594341Z","shell.execute_reply":"2024-12-05T12:40:11.452593Z"}},"outputs":[{"name":"stdout","text":"Collecting cirq\n  Downloading cirq-1.4.1-py3-none-any.whl.metadata (7.4 kB)\nCollecting cirq-aqt==1.4.1 (from cirq)\n  Downloading cirq_aqt-1.4.1-py3-none-any.whl.metadata (1.6 kB)\nCollecting cirq-core==1.4.1 (from cirq)\n  Downloading cirq_core-1.4.1-py3-none-any.whl.metadata (1.8 kB)\nCollecting cirq-google==1.4.1 (from cirq)\n  Downloading cirq_google-1.4.1-py3-none-any.whl.metadata (2.0 kB)\nCollecting cirq-ionq==1.4.1 (from cirq)\n  Downloading cirq_ionq-1.4.1-py3-none-any.whl.metadata (1.6 kB)\nCollecting cirq-pasqal==1.4.1 (from cirq)\n  Downloading cirq_pasqal-1.4.1-py3-none-any.whl.metadata (1.6 kB)\nCollecting cirq-rigetti==1.4.1 (from cirq)\n  Downloading cirq_rigetti-1.4.1-py3-none-any.whl.metadata (1.7 kB)\nCollecting cirq-web==1.4.1 (from cirq)\n  Downloading cirq_web-1.4.1-py3-none-any.whl.metadata (2.6 kB)\nRequirement already satisfied: requests~=2.18 in /opt/conda/lib/python3.10/site-packages (from cirq-aqt==1.4.1->cirq) (2.32.3)\nRequirement already satisfied: attrs>=21.3.0 in /opt/conda/lib/python3.10/site-packages (from cirq-core==1.4.1->cirq) (23.2.0)\nCollecting duet>=0.2.8 (from cirq-core==1.4.1->cirq)\n  Downloading duet-0.2.9-py3-none-any.whl.metadata (2.3 kB)\nRequirement already satisfied: matplotlib~=3.0 in /opt/conda/lib/python3.10/site-packages (from cirq-core==1.4.1->cirq) (3.7.5)\nRequirement already satisfied: networkx>=2.4 in /opt/conda/lib/python3.10/site-packages (from cirq-core==1.4.1->cirq) (3.3)\nRequirement already satisfied: numpy~=1.22 in /opt/conda/lib/python3.10/site-packages (from cirq-core==1.4.1->cirq) (1.26.4)\nRequirement already satisfied: pandas in /opt/conda/lib/python3.10/site-packages (from cirq-core==1.4.1->cirq) (2.2.3)\nCollecting sortedcontainers~=2.0 (from cirq-core==1.4.1->cirq)\n  Downloading sortedcontainers-2.4.0-py2.py3-none-any.whl.metadata (10 kB)\nRequirement already satisfied: scipy~=1.0 in /opt/conda/lib/python3.10/site-packages (from cirq-core==1.4.1->cirq) (1.14.1)\nRequirement already satisfied: sympy in /opt/conda/lib/python3.10/site-packages (from cirq-core==1.4.1->cirq) (1.13.1)\nRequirement already satisfied: typing-extensions>=4.2 in /opt/conda/lib/python3.10/site-packages (from cirq-core==1.4.1->cirq) (4.12.2)\nRequirement already satisfied: tqdm in /opt/conda/lib/python3.10/site-packages (from cirq-core==1.4.1->cirq) (4.66.4)\nRequirement already satisfied: google-api-core>=1.14.0 in /opt/conda/lib/python3.10/site-packages (from google-api-core[grpc]>=1.14.0->cirq-google==1.4.1->cirq) (2.11.1)\nRequirement already satisfied: proto-plus>=1.20.0 in /opt/conda/lib/python3.10/site-packages (from cirq-google==1.4.1->cirq) (1.23.0)\nRequirement already satisfied: protobuf<5.0.0,>=3.15.0 in /opt/conda/lib/python3.10/site-packages (from cirq-google==1.4.1->cirq) (3.20.3)\nCollecting pyquil<5.0.0,>=4.11.0 (from cirq-rigetti==1.4.1->cirq)\n  Downloading pyquil-4.14.3-py3-none-any.whl.metadata (10 kB)\nRequirement already satisfied: googleapis-common-protos<2.0.dev0,>=1.56.2 in /opt/conda/lib/python3.10/site-packages (from google-api-core>=1.14.0->google-api-core[grpc]>=1.14.0->cirq-google==1.4.1->cirq) (1.63.1)\nRequirement already satisfied: google-auth<3.0.dev0,>=2.14.1 in /opt/conda/lib/python3.10/site-packages (from google-api-core>=1.14.0->google-api-core[grpc]>=1.14.0->cirq-google==1.4.1->cirq) (2.30.0)\nRequirement already satisfied: grpcio<2.0dev,>=1.33.2 in /opt/conda/lib/python3.10/site-packages (from google-api-core[grpc]>=1.14.0->cirq-google==1.4.1->cirq) (1.64.1)\nRequirement already satisfied: grpcio-status<2.0.dev0,>=1.33.2 in /opt/conda/lib/python3.10/site-packages (from google-api-core[grpc]>=1.14.0->cirq-google==1.4.1->cirq) (1.48.0)\nRequirement already satisfied: contourpy>=1.0.1 in /opt/conda/lib/python3.10/site-packages (from matplotlib~=3.0->cirq-core==1.4.1->cirq) (1.2.1)\nRequirement already satisfied: cycler>=0.10 in /opt/conda/lib/python3.10/site-packages (from matplotlib~=3.0->cirq-core==1.4.1->cirq) (0.12.1)\nRequirement already satisfied: fonttools>=4.22.0 in /opt/conda/lib/python3.10/site-packages (from matplotlib~=3.0->cirq-core==1.4.1->cirq) (4.53.0)\nRequirement already satisfied: kiwisolver>=1.0.1 in /opt/conda/lib/python3.10/site-packages (from matplotlib~=3.0->cirq-core==1.4.1->cirq) (1.4.5)\nRequirement already satisfied: packaging>=20.0 in /opt/conda/lib/python3.10/site-packages (from matplotlib~=3.0->cirq-core==1.4.1->cirq) (21.3)\nRequirement already satisfied: pillow>=6.2.0 in /opt/conda/lib/python3.10/site-packages (from matplotlib~=3.0->cirq-core==1.4.1->cirq) (10.3.0)\nRequirement already satisfied: pyparsing>=2.3.1 in /opt/conda/lib/python3.10/site-packages (from matplotlib~=3.0->cirq-core==1.4.1->cirq) (3.1.2)\nRequirement already satisfied: python-dateutil>=2.7 in /opt/conda/lib/python3.10/site-packages (from matplotlib~=3.0->cirq-core==1.4.1->cirq) (2.9.0.post0)\nRequirement already satisfied: deprecated<2.0.0,>=1.2.14 in /opt/conda/lib/python3.10/site-packages (from pyquil<5.0.0,>=4.11.0->cirq-rigetti==1.4.1->cirq) (1.2.14)\nRequirement already satisfied: matplotlib-inline<0.2.0,>=0.1.7 in /opt/conda/lib/python3.10/site-packages (from pyquil<5.0.0,>=4.11.0->cirq-rigetti==1.4.1->cirq) (0.1.7)\nCollecting packaging>=20.0 (from matplotlib~=3.0->cirq-core==1.4.1->cirq)\n  Downloading packaging-23.2-py3-none-any.whl.metadata (3.2 kB)\nCollecting qcs-sdk-python>=0.20.1 (from pyquil<5.0.0,>=4.11.0->cirq-rigetti==1.4.1->cirq)\n  Downloading qcs_sdk_python-0.21.4-cp310-cp310-manylinux_2_28_x86_64.whl.metadata (7.0 kB)\nCollecting quil>=0.11.2 (from pyquil<5.0.0,>=4.11.0->cirq-rigetti==1.4.1->cirq)\n  Downloading quil-0.13.2-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (1.8 kB)\nCollecting rpcq<4.0.0,>=3.11.0 (from pyquil<5.0.0,>=4.11.0->cirq-rigetti==1.4.1->cirq)\n  Downloading rpcq-3.11.0.tar.gz (45 kB)\n\u001b[2K     \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m45.6/45.6 kB\u001b[0m \u001b[31m1.3 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n\u001b[?25h  Preparing metadata (setup.py) ... \u001b[?25ldone\n\u001b[?25hCollecting types-deprecated<2.0.0.0,>=1.2.9.3 (from pyquil<5.0.0,>=4.11.0->cirq-rigetti==1.4.1->cirq)\n  Downloading types_Deprecated-1.2.15.20241117-py3-none-any.whl.metadata (1.8 kB)\nRequirement already satisfied: charset-normalizer<4,>=2 in /opt/conda/lib/python3.10/site-packages (from requests~=2.18->cirq-aqt==1.4.1->cirq) (3.3.2)\nRequirement already satisfied: idna<4,>=2.5 in /opt/conda/lib/python3.10/site-packages (from requests~=2.18->cirq-aqt==1.4.1->cirq) (3.7)\nRequirement already satisfied: urllib3<3,>=1.21.1 in /opt/conda/lib/python3.10/site-packages (from requests~=2.18->cirq-aqt==1.4.1->cirq) (1.26.18)\nRequirement already satisfied: certifi>=2017.4.17 in /opt/conda/lib/python3.10/site-packages (from requests~=2.18->cirq-aqt==1.4.1->cirq) (2024.6.2)\nRequirement already satisfied: pytz>=2020.1 in /opt/conda/lib/python3.10/site-packages (from pandas->cirq-core==1.4.1->cirq) (2024.1)\nRequirement already satisfied: tzdata>=2022.7 in /opt/conda/lib/python3.10/site-packages (from pandas->cirq-core==1.4.1->cirq) (2024.1)\nRequirement already satisfied: mpmath<1.4,>=1.1.0 in /opt/conda/lib/python3.10/site-packages (from sympy->cirq-core==1.4.1->cirq) (1.3.0)\nRequirement already satisfied: wrapt<2,>=1.10 in /opt/conda/lib/python3.10/site-packages (from deprecated<2.0.0,>=1.2.14->pyquil<5.0.0,>=4.11.0->cirq-rigetti==1.4.1->cirq) (1.16.0)\nRequirement already satisfied: cachetools<6.0,>=2.0.0 in /opt/conda/lib/python3.10/site-packages (from google-auth<3.0.dev0,>=2.14.1->google-api-core>=1.14.0->google-api-core[grpc]>=1.14.0->cirq-google==1.4.1->cirq) (4.2.4)\nRequirement already satisfied: pyasn1-modules>=0.2.1 in /opt/conda/lib/python3.10/site-packages (from google-auth<3.0.dev0,>=2.14.1->google-api-core>=1.14.0->google-api-core[grpc]>=1.14.0->cirq-google==1.4.1->cirq) (0.4.0)\nRequirement already satisfied: rsa<5,>=3.1.4 in /opt/conda/lib/python3.10/site-packages (from google-auth<3.0.dev0,>=2.14.1->google-api-core>=1.14.0->google-api-core[grpc]>=1.14.0->cirq-google==1.4.1->cirq) (4.9)\nRequirement already satisfied: traitlets in /opt/conda/lib/python3.10/site-packages (from matplotlib-inline<0.2.0,>=0.1.7->pyquil<5.0.0,>=4.11.0->cirq-rigetti==1.4.1->cirq) (5.14.3)\nRequirement already satisfied: six>=1.5 in /opt/conda/lib/python3.10/site-packages (from python-dateutil>=2.7->matplotlib~=3.0->cirq-core==1.4.1->cirq) (1.16.0)\nCollecting qcs-api-client-common>=0.10.0 (from qcs-sdk-python>=0.20.1->pyquil<5.0.0,>=4.11.0->cirq-rigetti==1.4.1->cirq)\n  Downloading qcs_api_client_common-0.10.0-cp310-cp310-manylinux_2_28_x86_64.whl.metadata (2.5 kB)\nRequirement already satisfied: msgpack<2.0,>=0.6 in /opt/conda/lib/python3.10/site-packages (from rpcq<4.0.0,>=3.11.0->pyquil<5.0.0,>=4.11.0->cirq-rigetti==1.4.1->cirq) (1.0.8)\nCollecting python-rapidjson (from rpcq<4.0.0,>=3.11.0->pyquil<5.0.0,>=4.11.0->cirq-rigetti==1.4.1->cirq)\n  Downloading python_rapidjson-1.20-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (22 kB)\nRequirement already satisfied: pyzmq>=17 in /opt/conda/lib/python3.10/site-packages (from rpcq<4.0.0,>=3.11.0->pyquil<5.0.0,>=4.11.0->cirq-rigetti==1.4.1->cirq) (26.0.3)\nRequirement already satisfied: ruamel.yaml in /opt/conda/lib/python3.10/site-packages (from rpcq<4.0.0,>=3.11.0->pyquil<5.0.0,>=4.11.0->cirq-rigetti==1.4.1->cirq) (0.18.6)\nRequirement already satisfied: pyasn1<0.7.0,>=0.4.6 in /opt/conda/lib/python3.10/site-packages (from pyasn1-modules>=0.2.1->google-auth<3.0.dev0,>=2.14.1->google-api-core>=1.14.0->google-api-core[grpc]>=1.14.0->cirq-google==1.4.1->cirq) (0.6.0)\nRequirement already satisfied: grpc-interceptor>=0.15.0 in /opt/conda/lib/python3.10/site-packages (from qcs-api-client-common>=0.10.0->qcs-sdk-python>=0.20.1->pyquil<5.0.0,>=4.11.0->cirq-rigetti==1.4.1->cirq) (0.15.4)\nRequirement already satisfied: httpx>=0.27.0 in /opt/conda/lib/python3.10/site-packages (from qcs-api-client-common>=0.10.0->qcs-sdk-python>=0.20.1->pyquil<5.0.0,>=4.11.0->cirq-rigetti==1.4.1->cirq) (0.27.0)\nRequirement already satisfied: ruamel.yaml.clib>=0.2.7 in /opt/conda/lib/python3.10/site-packages (from ruamel.yaml->rpcq<4.0.0,>=3.11.0->pyquil<5.0.0,>=4.11.0->cirq-rigetti==1.4.1->cirq) (0.2.8)\nRequirement already satisfied: anyio in /opt/conda/lib/python3.10/site-packages (from httpx>=0.27.0->qcs-api-client-common>=0.10.0->qcs-sdk-python>=0.20.1->pyquil<5.0.0,>=4.11.0->cirq-rigetti==1.4.1->cirq) (4.4.0)\nRequirement already satisfied: httpcore==1.* in /opt/conda/lib/python3.10/site-packages (from httpx>=0.27.0->qcs-api-client-common>=0.10.0->qcs-sdk-python>=0.20.1->pyquil<5.0.0,>=4.11.0->cirq-rigetti==1.4.1->cirq) (1.0.5)\nRequirement already satisfied: sniffio in /opt/conda/lib/python3.10/site-packages (from httpx>=0.27.0->qcs-api-client-common>=0.10.0->qcs-sdk-python>=0.20.1->pyquil<5.0.0,>=4.11.0->cirq-rigetti==1.4.1->cirq) (1.3.1)\nRequirement already satisfied: h11<0.15,>=0.13 in /opt/conda/lib/python3.10/site-packages (from httpcore==1.*->httpx>=0.27.0->qcs-api-client-common>=0.10.0->qcs-sdk-python>=0.20.1->pyquil<5.0.0,>=4.11.0->cirq-rigetti==1.4.1->cirq) (0.14.0)\nRequirement already satisfied: exceptiongroup>=1.0.2 in /opt/conda/lib/python3.10/site-packages (from anyio->httpx>=0.27.0->qcs-api-client-common>=0.10.0->qcs-sdk-python>=0.20.1->pyquil<5.0.0,>=4.11.0->cirq-rigetti==1.4.1->cirq) (1.2.0)\nDownloading cirq-1.4.1-py3-none-any.whl (8.0 kB)\nDownloading cirq_aqt-1.4.1-py3-none-any.whl (30 kB)\nDownloading cirq_core-1.4.1-py3-none-any.whl (1.9 MB)\n\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m1.9/1.9 MB\u001b[0m \u001b[31m27.8 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0ma \u001b[36m0:00:01\u001b[0m\n\u001b[?25hDownloading cirq_google-1.4.1-py3-none-any.whl (532 kB)\n\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m532.7/532.7 kB\u001b[0m \u001b[31m22.1 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n\u001b[?25hDownloading cirq_ionq-1.4.1-py3-none-any.whl (60 kB)\n\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m60.5/60.5 kB\u001b[0m \u001b[31m2.7 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n\u001b[?25hDownloading cirq_pasqal-1.4.1-py3-none-any.whl (31 kB)\nDownloading cirq_rigetti-1.4.1-py3-none-any.whl (69 kB)\n\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m69.3/69.3 kB\u001b[0m \u001b[31m3.0 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n\u001b[?25hDownloading cirq_web-1.4.1-py3-none-any.whl (596 kB)\n\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m596.5/596.5 kB\u001b[0m \u001b[31m26.6 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n\u001b[?25hDownloading duet-0.2.9-py3-none-any.whl (29 kB)\nDownloading pyquil-4.14.3-py3-none-any.whl (202 kB)\n\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m202.8/202.8 kB\u001b[0m \u001b[31m11.5 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n\u001b[?25hDownloading sortedcontainers-2.4.0-py2.py3-none-any.whl (29 kB)\nDownloading packaging-23.2-py3-none-any.whl (53 kB)\n\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m53.0/53.0 kB\u001b[0m \u001b[31m2.3 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n\u001b[?25hDownloading qcs_sdk_python-0.21.4-cp310-cp310-manylinux_2_28_x86_64.whl (6.4 MB)\n\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m6.4/6.4 MB\u001b[0m \u001b[31m70.3 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m:00:01\u001b[0m00:01\u001b[0m\n\u001b[?25hDownloading quil-0.13.2-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (2.2 MB)\n\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m2.2/2.2 MB\u001b[0m \u001b[31m54.8 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m:00:01\u001b[0m\n\u001b[?25hDownloading types_Deprecated-1.2.15.20241117-py3-none-any.whl (3.8 kB)\nDownloading qcs_api_client_common-0.10.0-cp310-cp310-manylinux_2_28_x86_64.whl (2.9 MB)\n\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m2.9/2.9 MB\u001b[0m \u001b[31m56.9 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m:00:01\u001b[0m\n\u001b[?25hDownloading python_rapidjson-1.20-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (1.7 MB)\n\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m1.7/1.7 MB\u001b[0m \u001b[31m45.5 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m:00:01\u001b[0m\n\u001b[?25hBuilding wheels for collected packages: rpcq\n  Building wheel for rpcq (setup.py) ... \u001b[?25ldone\n\u001b[?25h  Created wheel for rpcq: filename=rpcq-3.11.0-py3-none-any.whl size=45967 sha256=276f64b57fe426c4522fea2801d1cd669e2e85293f6b593a3f149b677aa1561f\n  Stored in directory: /root/.cache/pip/wheels/91/97/0a/b384d94155345284ec00712ab814cbfb5bc2b631080b4fa4c3\nSuccessfully built rpcq\nInstalling collected packages: sortedcontainers, types-deprecated, quil, python-rapidjson, packaging, duet, rpcq, qcs-api-client-common, cirq-core, qcs-sdk-python, cirq-web, cirq-pasqal, cirq-ionq, cirq-aqt, pyquil, cirq-google, cirq-rigetti, cirq\n  Attempting uninstall: packaging\n    Found existing installation: packaging 21.3\n    Uninstalling packaging-21.3:\n      Successfully uninstalled packaging-21.3\n\u001b[31mERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.\ngoogle-cloud-bigquery 2.34.4 requires packaging<22.0dev,>=14.3, but you have packaging 23.2 which is incompatible.\njupyterlab 4.3.1 requires jupyter-lsp>=2.0.0, but you have jupyter-lsp 1.5.1 which is incompatible.\njupyterlab-lsp 5.1.0 requires jupyter-lsp>=2.0.0, but you have jupyter-lsp 1.5.1 which is incompatible.\nlibpysal 4.9.2 requires shapely>=2.0.1, but you have shapely 1.8.5.post1 which is incompatible.\nmlxtend 0.23.3 requires scikit-learn>=1.3.1, but you have scikit-learn 1.2.2 which is incompatible.\nplotnine 0.14.3 requires matplotlib>=3.8.0, but you have matplotlib 3.7.5 which is incompatible.\ntensorflow 2.16.1 requires ml-dtypes~=0.3.1, but you have ml-dtypes 0.5.0 which is incompatible.\nthinc 8.3.2 requires numpy<2.1.0,>=2.0.0; python_version >= \"3.9\", but you have numpy 1.26.4 which is incompatible.\nydata-profiling 4.12.0 requires scipy<1.14,>=1.4.1, but you have scipy 1.14.1 which is incompatible.\u001b[0m\u001b[31m\n\u001b[0mSuccessfully installed cirq-1.4.1 cirq-aqt-1.4.1 cirq-core-1.4.1 cirq-google-1.4.1 cirq-ionq-1.4.1 cirq-pasqal-1.4.1 cirq-rigetti-1.4.1 cirq-web-1.4.1 duet-0.2.9 packaging-23.2 pyquil-4.14.3 python-rapidjson-1.20 qcs-api-client-common-0.10.0 qcs-sdk-python-0.21.4 quil-0.13.2 rpcq-3.11.0 sortedcontainers-2.4.0 types-deprecated-1.2.15.20241117\n","output_type":"stream"}],"execution_count":57},{"cell_type":"code","source":"import cirq\nimport tensorflow as tf","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2024-12-05T12:40:54.624322Z","iopub.execute_input":"2024-12-05T12:40:54.625175Z","iopub.status.idle":"2024-12-05T12:40:54.630308Z","shell.execute_reply.started":"2024-12-05T12:40:54.625131Z","shell.execute_reply":"2024-12-05T12:40:54.629153Z"}},"outputs":[],"execution_count":59},{"cell_type":"code","source":"def quantum_circuit(data):\n    qubit=[cirq.GridQubit(0,i) for i in range(13)]\n    circuit=cirq.Circuit()\n    for i,value in enumerate(data):\n        q=i%len(qubit)\n        circuit.append(cirq.H(qubit[q]))\n        circuit.append(cirq.rx(value*np.pi).on(qubit[q]))\n        circuit.append(cirq.ry(value*np.pi).on(qubit[q]))\n        if (i+1)%len(qubit)<len(qubit):\n            circuit.append(cirq.CNOT(qubit[q],qubit[(i+1)%len(qubit)]))\n            circuit.append(cirq.measure(qubit[q],key=f\"qubit_{i}\"))\n    return circuit,qubit\ndef run_quantum_circuit(data):\n    results=[]\n    simulator=cirq.Simulator()\n    for inputs_data in data:\n        circuit,qubit=quantum_circuit(inputs_data)\n        result=simulator.run(circuit)\n        measure0=[]\n        for i in range(len(qubit)):\n            measure0.append(result.measurements[f\"qubit_{i}\"].flatten()[0])\n        results.append(measure0)\n    return np.array(results,dtype=np.float32)\nclass QuantumLayer(Layer):\n    def __init__(self):\n        super(QuantumLayer,self).__init__()\n    def call(self,inputs):\n        outputs=tf.numpy_function(run_quantum_circuit,[inputs],tf.float32)\n        outputs.set_shape(inputs.shape)\n        return outputs\nmodel2=Sequential([\n    QuantumLayer(),\n    Dense(32,activation=\"relu\",kernel_initializer=HeNormal(),kernel_regularizer=l2(0.01)),\n    Dense(64,kernel_initializer=HeNormal(),kernel_regularizer=l2(0.01)),\n    LeakyReLU(0.08),\n    Dense(128,kernel_initializer=GlorotUniform(),kernel_regularizer=l2(0.01)),\n    LeakyReLU(0.08),\n    Dense(228,kernel_initializer=GlorotUniform(),kernel_regularizer=l2(0.01)),\n    LeakyReLU(0.08),\n    Dense(528,kernel_initializer=GlorotUniform(),kernel_regularizer=l2(0.01)),\n    LeakyReLU(0.08),\n    Dense(1028,kernel_initializer=GlorotUniform(),kernel_regularizer=l2(0.01)),\n    LeakyReLU(0.08),\n    Dense(1)\n])\nmodel2.compile(optimizer=\"adam\",loss=\"mae\",metrics=[\"mae\"])\nmodel2.fit(X_train,Y_train,epochs=15,validation_data=(X_test,Y_test),batch_size=32)","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2024-12-05T13:12:28.245776Z","iopub.execute_input":"2024-12-05T13:12:28.246218Z","iopub.status.idle":"2024-12-05T15:04:26.551343Z","shell.execute_reply.started":"2024-12-05T13:12:28.246181Z","shell.execute_reply":"2024-12-05T15:04:26.550150Z"}},"outputs":[{"name":"stdout","text":"Epoch 1/15\n\u001b[1m551/551\u001b[0m \u001b[32m━━━━━━━━━━━━━━━━━━━━\u001b[0m\u001b[37m\u001b[0m \u001b[1m447s\u001b[0m 806ms/step - loss: 6671.9429 - mae: 6657.4648 - val_loss: 5535.4644 - val_mae: 5522.0640\nEpoch 2/15\n\u001b[1m551/551\u001b[0m \u001b[32m━━━━━━━━━━━━━━━━━━━━\u001b[0m\u001b[37m\u001b[0m \u001b[1m444s\u001b[0m 806ms/step - loss: 5461.9761 - mae: 5449.2368 - val_loss: 5934.9341 - val_mae: 5922.7314\nEpoch 3/15\n\u001b[1m551/551\u001b[0m \u001b[32m━━━━━━━━━━━━━━━━━━━━\u001b[0m\u001b[37m\u001b[0m \u001b[1m450s\u001b[0m 817ms/step - loss: 5539.4004 - mae: 5527.9629 - val_loss: 5540.0039 - val_mae: 5528.8125\nEpoch 4/15\n\u001b[1m551/551\u001b[0m \u001b[32m━━━━━━━━━━━━━━━━━━━━\u001b[0m\u001b[37m\u001b[0m \u001b[1m450s\u001b[0m 817ms/step - loss: 5455.6836 - mae: 5444.9121 - val_loss: 5399.7407 - val_mae: 5389.3491\nEpoch 5/15\n\u001b[1m551/551\u001b[0m \u001b[32m━━━━━━━━━━━━━━━━━━━━\u001b[0m\u001b[37m\u001b[0m \u001b[1m449s\u001b[0m 816ms/step - loss: 5432.4829 - mae: 5422.1851 - val_loss: 5523.3281 - val_mae: 5513.5850\nEpoch 6/15\n\u001b[1m551/551\u001b[0m \u001b[32m━━━━━━━━━━━━━━━━━━━━\u001b[0m\u001b[37m\u001b[0m \u001b[1m447s\u001b[0m 812ms/step - loss: 5455.1221 - mae: 5445.1201 - val_loss: 5412.8193 - val_mae: 5403.0464\nEpoch 7/15\n\u001b[1m551/551\u001b[0m \u001b[32m━━━━━━━━━━━━━━━━━━━━\u001b[0m\u001b[37m\u001b[0m \u001b[1m454s\u001b[0m 824ms/step - loss: 5446.2759 - mae: 5436.5049 - val_loss: 5427.1865 - val_mae: 5417.7002\nEpoch 8/15\n\u001b[1m551/551\u001b[0m \u001b[32m━━━━━━━━━━━━━━━━━━━━\u001b[0m\u001b[37m\u001b[0m \u001b[1m447s\u001b[0m 812ms/step - loss: 5460.5127 - mae: 5450.9810 - val_loss: 5397.1211 - val_mae: 5387.7173\nEpoch 9/15\n\u001b[1m551/551\u001b[0m \u001b[32m━━━━━━━━━━━━━━━━━━━━\u001b[0m\u001b[37m\u001b[0m \u001b[1m446s\u001b[0m 809ms/step - loss: 5422.9858 - mae: 5413.6196 - val_loss: 5399.5195 - val_mae: 5390.2441\nEpoch 10/15\n\u001b[1m551/551\u001b[0m \u001b[32m━━━━━━━━━━━━━━━━━━━━\u001b[0m\u001b[37m\u001b[0m \u001b[1m451s\u001b[0m 820ms/step - loss: 5469.5273 - mae: 5460.3418 - val_loss: 5424.1338 - val_mae: 5415.1577\nEpoch 11/15\n\u001b[1m551/551\u001b[0m \u001b[32m━━━━━━━━━━━━━━━━━━━━\u001b[0m\u001b[37m\u001b[0m \u001b[1m448s\u001b[0m 813ms/step - loss: 5399.8257 - mae: 5390.7666 - val_loss: 5419.8413 - val_mae: 5410.7871\nEpoch 12/15\n\u001b[1m551/551\u001b[0m \u001b[32m━━━━━━━━━━━━━━━━━━━━\u001b[0m\u001b[37m\u001b[0m \u001b[1m446s\u001b[0m 810ms/step - loss: 5415.0098 - mae: 5406.0835 - val_loss: 5408.7822 - val_mae: 5399.8398\nEpoch 13/15\n\u001b[1m551/551\u001b[0m \u001b[32m━━━━━━━━━━━━━━━━━━━━\u001b[0m\u001b[37m\u001b[0m \u001b[1m445s\u001b[0m 809ms/step - loss: 5454.2070 - mae: 5445.3701 - val_loss: 5394.7324 - val_mae: 5385.9385\nEpoch 14/15\n\u001b[1m551/551\u001b[0m \u001b[32m━━━━━━━━━━━━━━━━━━━━\u001b[0m\u001b[37m\u001b[0m \u001b[1m446s\u001b[0m 809ms/step - loss: 5401.0640 - mae: 5392.2690 - val_loss: 5388.9170 - val_mae: 5380.3042\nEpoch 15/15\n\u001b[1m551/551\u001b[0m \u001b[32m━━━━━━━━━━━━━━━━━━━━\u001b[0m\u001b[37m\u001b[0m \u001b[1m448s\u001b[0m 813ms/step - loss: 5427.7754 - mae: 5419.0820 - val_loss: 5392.9692 - val_mae: 5384.4727\n","output_type":"stream"},{"execution_count":67,"output_type":"execute_result","data":{"text/plain":"<keras.src.callbacks.history.History at 0x7a1e4d91a290>"},"metadata":{}}],"execution_count":67},{"cell_type":"code","source":"model2.predict(X_test.iloc[:10,:])","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2024-12-05T15:10:30.229928Z","iopub.execute_input":"2024-12-05T15:10:30.231195Z","iopub.status.idle":"2024-12-05T15:10:30.565664Z","shell.execute_reply.started":"2024-12-05T15:10:30.231136Z","shell.execute_reply":"2024-12-05T15:10:30.564299Z"}},"outputs":[{"name":"stdout","text":"\u001b[1m1/1\u001b[0m \u001b[32m━━━━━━━━━━━━━━━━━━━━\u001b[0m\u001b[37m\u001b[0m \u001b[1m0s\u001b[0m 263ms/step\n","output_type":"stream"},{"execution_count":76,"output_type":"execute_result","data":{"text/plain":"array([[10209.737],\n       [ 9414.887],\n       [ 9563.236],\n       [10702.491],\n       [10620.666],\n       [10549.229],\n       [ 9878.156],\n       [ 9309.834],\n       [ 9671.833],\n       [11044.105]], dtype=float32)"},"metadata":{}}],"execution_count":76},{"cell_type":"code","source":"Y_test[:10]","metadata":{"trusted":true,"execution":{"iopub.status.busy":"2024-12-05T15:10:37.594569Z","iopub.execute_input":"2024-12-05T15:10:37.595522Z","iopub.status.idle":"2024-12-05T15:10:37.603981Z","shell.execute_reply.started":"2024-12-05T15:10:37.595479Z","shell.execute_reply":"2024-12-05T15:10:37.602568Z"}},"outputs":[{"execution_count":77,"output_type":"execute_result","data":{"text/plain":"11429     9030\n14802    20627\n8323     21046\n16757    17984\n12685     4653\n18783    19110\n3827      9463\n221      17648\n10563     7426\n10492    13694\nName: Strain, dtype: int64"},"metadata":{}}],"execution_count":77},{"cell_type":"code","source":"","metadata":{"trusted":true},"outputs":[],"execution_count":null},{"cell_type":"code","source":"","metadata":{"trusted":true},"outputs":[],"execution_count":null},{"cell_type":"code","source":"","metadata":{"trusted":true},"outputs":[],"execution_count":null},{"cell_type":"code","source":"","metadata":{"trusted":true},"outputs":[],"execution_count":null}]}